docs/dev/Lua API.rst

```diff
@@ -3928,6 +3928,211 @@ Examples:
+
+  :year: A year. Defaults to zero.
+  :year_tick: A year tick. Defaults to zero.
+  :ticks_per_day: The rate at which time passes in a day.
+                  Defaults to DWARF/fortress mode.
+                  note: if ticks_per_day is given a value <= 1,
+                  it will default to fortress mode.
+  Examples::
+
+    -- create a new instance with year and year_tick set to
+    -- the current game time.
+    local dateTime = DateTime.now()
+    -- construct and set year to 1, year_tick to 1
+    dateTime = DateTime{ year = 1, year_tick = 1 }
+    -- set ticks_per_day to adventure mode
+    dateTime = DateTime{ ticks_per_day = df.game_mode.ADVENTURE }
+
+    local dateTimeOne = DateTime{ year = 1, year_tick = 1}
+    local dateTimeTwo = DateTime{ year = 1, year_tick = 2}
+    local deltaDateTime = dateTimeTwo - dateTimeOne
+    deltaDateTime = dateTimeOne + dateTimeTwo
+    if (dateTimeOne < deltaDateTime) then print('is less than') end
+
+* ``dateTime:year()``
+
+  Returns an integer pair: *year*, *year tick count*.
+
+* ``dateTime:setYear(year)``
+
+  Sets the year. Returns *DateTime*.
+
+* ``dateTime:setYearTick(ticks)``
+
+  Sets year_tick to ticks, normalizing if needed.
+  It is advised to use this function instead of
+  assigning to year_tick directly due to normalization.
+  returns *DateTime*.
+
+* ``dateTime:dayOfYear()``
+
+  Returns an integer pair: *the current day of the year starting from one*, *day tick count*.
+
+* ``dateTime:setDayOfYear(day)``
+
+  Sets the day of year starting from one.
+  returns *DateTime*.
+
+* ``dateTime:dayOfMonth()``
+
+  Returns an integer pair: *the current day of the month starting from one*, *day tick count*.
+
+* ``dateTime:setDayOfMonth(day[, month])``
+
+  Sets the day of month. If month is not given, defaults to current month.
+  returns *DateTime*.
+
+* ``dateTime:dayOfMonthWithSuffix()``
+
+  Returns *day of month, as a string, including its ordinal suffix*.
+
+* ``dateTime:month()``
+
+  Returns *current month, as a number, starting from one*.
+
+* ``dateTime:setMonth(month)``
+
+  Sets the current month starting from one.
+  Returns *DateTime*.
+
+* ``dateTime:nameOfMonth()``
+
+  Returns *name of the current month*.
+
+* ``dateTime:dayOfYearTick()``
+
+  Returns *days from ticks since beginning of current year starting from zero*.
+
+* ``dateTime:dayOfMonthTick()``
+
+  Returns *days from ticks since the beginning of current month starting from zero*.
+
+* ``dateTime:dayTick()``
+
+  Returns *ticks since beginning of the current day starting from zero*.
+
+* ``dateTime:setDayTick(ticks)``
+
+  Sets the current day's tick count.
+
+* ``dateTime:monthOfYearTick()``
+
+  Returns *months from ticks since the beginning of the current year starting from zero*.
+
+* ``dateTime:monthTick()``
+
+  Returns *ticks since the beginning of a month*.
+
+* ``dateTime:daysToTicks(days)``
+
+  Returns *days converted to ticks*.
+
+* ``dateTime:monthsToTicks(months)``
```

Comment:
these look like class functions instead of instance functions. Should they be documented as `DateTime.` instead of `dateTime:`?

Reply:
These are calendrical functions that use the current tick rate for a particular instance (either by game mode or custom tick rate).  They could be class functions if we want all instances to share the same tick rate.  There used to be a mutable TICKS_PER_DAY module scoped variable for this, but taken out when custom tick rate logic was added.

Reply:
that's a good point. I forgot that the tick rate could differ.

How about making them class functions, but taking the mode as an optional parameter, defaulting to the current mode like the constructor does?

Reply:
Fine with me; however, if we treat mode as a boolean, we won't be able to support a custom tick rate. Tick rate will either be fort or adv.

Reply:
Or we could pass in an optional object and get the mode and stuff from there.

Reply:
Either a boolean or a documented constant are fine, I think. The tick rate of the various modes is *not* moddable afaik. even `timestream` doesn't change the ticks per month -- it just advances the ticks faster than they would normally advance.